js/api/Surveys.js

```diff
@@ -216,16 +226,17 @@ export function loadSurveys(callback) {
 /**
  * fetches remote data for surveys and invitations
  *
+ * @param {object}   options, options object to be passed to loadSurveys
  * @param {function} done, the callback for when the async operations are done
  */
-export function loadSurveyList(done) {
+export function loadSurveyList(options = {}, done) {
   async.auto({
-    surveys: (cb) => {
-      loadSurveys(cb);
-    },
     invitations: (cb) => {
       loadInvitations(cb);
     },
+    surveys: ['invitations', (cb) => {
```

Comment:
Here is a couple of suggestions:
a) Limit the initial number of surveys then implement either a [Load More] button as the last element of the list or infinite scrolling.
b) Fetch surveys, and form and question counts on the SurveyListPage. Then fetch/load the relations as they are needed, possibly upon acceptance of the SurveyDetailPage.